### PR TITLE
tpm_with_hlk: New case to test vTPM with HLK

### PR DESCRIPTION
**Pull request by YongxueHong**
ID: 1861615

Signed-off-by: Yongxue Hong yhong@redhat.com

### CODE REVIEW

**Comment by YongxueHong** [comment]
Hi, @qcheng-redhat Please help me to review, thanks.

**Comment by qcheng-redhat** [comment]
Test result for Win10 guest:
python3 ConfigTest.py --testcase=tpm_with_hlk --machine=q35 --firmware=ovmf --guestname=Win10 --clone=no

(1/1) type_specific.io-github-autotest-qemu.tpm_with_hlk: ERROR: Found task error messages:['Task error message : Test name: TPM - Auxiliary Test, Task name: Run Test, Cause : Task "Run Test" is Marked Failed From the LogFile', 'Task error message : Test name: TPM 2.0 EK Certificate Tests, Task name: Run Test, Cause : ... (10513.46 s)

The two test cases TPM - Auxiliary Test and TPM 2.0 EK Certificate Tests failed are expected, due to product issue.

**Comment by qcheng-redhat** [comment]
Test result for Win2019 guest:

python3 ConfigTest.py --testcase=tpm_with_hlk --machine=q35 --firmware=ovmf --guestname=Win2019 --clone=no 

(1/1) type_specific.io-github-autotest-qemu.tpm_with_hlk: ERROR: Found task error messages:['Task error message : Test name: TPM - Auxiliary Test, Task name: Run Test, Cause : Task "Run Test" is Marked Failed From the LogFile', 'Task error message : Test name: TPM 2.0 EK Certificate Tests, Task name: Run Test, Cause : ... (11158.46 s)

Failed test cases are expected. Detailed explanation for these two failed test cases, please refer to 
https://bugzilla.redhat.com/show_bug.cgi?id=1744045#c44

File: qemu/tests/tpm_with_hlk.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            vm_client = vm
+
+    install_hlk_client(vm_client, vm_server)
+
+    pool_name = params.get('hlk_pool_name')
+    project_name = params.get('hlk_project_name')
+    target_name = params.get('hlk_target_name')
+    tests_name = [name for name in params.get('hlk_target_tests_name').split(';')]
+    hlk_server = HLKServer(test, vm_server)
+    hlk_server.simple_run_test(pool_name, project_name, target_name,
```
Comment (qcheng-redhat) [inline qemu/tests/tpm_with_hlk.py:49]: timeout set to 24000 to ensure the test run on some old hosts don't reach timeout.

File: qemu/tests/tpm_with_hlk.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                               tests_name, timeout=24000, step=600)
```
Comment (qcheng-redhat) [inline qemu/tests/tpm_with_hlk.py:50]: Extend timeout to 24000, make sure to run the test on some old hosts don't reach timeout

Comment (YongxueHong) [reply qemu/tests/tpm_with_hlk.py:50]: Thanks, have updated it.

File: provider/win_hlk_suite.py [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+        :param step: Time to sleep between attempts in seconds.
+        :type step: int
+        """
+        default_pool = self.get_default_pool()[0]
+        machine_name = default_pool['machine_name']
+        self.create_pool(pool_name)
+        self.move_machine_from_default_pool(machine_name, pool_name)
+        self.set_machine_state(machine_name, pool_name, STATE_READY)
+        # FIXME: Sleep to provide buffer to execute next steps.
+        time.sleep(60)
```
Comment (qcheng-redhat) [inline provider/win_hlk_suite.py:500]: On some old hosts, 30 is not enough, extend to 60.

Comment (YongxueHong) [reply provider/win_hlk_suite.py:500]: Thanks, have updated.

File: provider/win_hlk_suite.py [hunk 4]
```diff
@@ -0,0 +1,587 @@
...
+    def set_machine_state(self, machine_name, pool_name, state, timeout=360):
+        """
+        Set machine state to "Ready" or "NotReady"
+
+        :param machine_name: Machine name.
+        :type machine_name: str
+        :param pool_name: Pool name.
+        :type pool_name: str
+        :param state: State to be set.
+        :type state: STATE_READY or STATE_READY
```
Comment (qcheng-redhat) [inline provider/win_hlk_suite.py:94]: STATE_READY or STATE_NOT_READY  ?

Comment (YongxueHong) [reply provider/win_hlk_suite.py]: Thanks, updated.

**Comment by qcheng-redhat** [comment]
I also tested with PASS scenarios:

# python3 scan_results.py 
CaseName                                 Status          StartTime       EndTime         Result     TimeElapsed
'1-type_specific.io-github-autotest-qemu.tpm_with_hlk' Finshed 2020-12-15 03:15:08 2020-12-15 06:35:01 PASS 11992.544871
Summary: 
Finshed=1, PASS=1

LGTM.

Thanks.

**Pull request merged**